exchanges/poloniex/poloniex.go

```diff
@@ -3,933 +3,1124 @@ package poloniex
+		return idPath + orderID, nil
+	case clientOrderID != "":
+		return clientIDPath + clientOrderID, nil
+	default:
+		return "", order.ErrOrderIDNotSet
+	}
+}
 
-	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, poloniexMarginPositionClose, values, &result)
+// CancelReplaceSmartOrder cancel an existing untriggered smart order and place a new smart order on the same symbol with details from existing smart order unless amended by new parameters
+func (e *Exchange) CancelReplaceSmartOrder(ctx context.Context, arg *CancelReplaceSmartOrderRequest) (*CancelReplaceSmartOrder, error) {
+	path, err := orderPath(arg.OrderID, "/smartorders/", arg.ClientOrderID, "/smartorders/cid:")
 	if err != nil {
-		return false, err
+		return nil, err
 	}
-
-	if result.Success == 0 {
-		return false, errors.New(result.Error)
+	var resp *CancelReplaceSmartOrder
+	if err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, sCreateReplaceSmartOrdersEPL, http.MethodPut, path, nil, arg, &resp); err != nil {
+		return nil, fmt.Errorf("%w: %w", order.ErrCancelFailed, err)
 	}
-
-	return true, nil
+	return resp, nil
 }
 
-// CreateLoanOffer places a loan offer on the exchange
-func (e *Exchange) CreateLoanOffer(ctx context.Context, ccy string, amount, rate float64, duration int, autoRenew bool) (int64, error) {
-	values := url.Values{}
-	values.Set("currency", ccy)
-	values.Set("amount", strconv.FormatFloat(amount, 'f', -1, 64))
-	values.Set("duration", strconv.Itoa(duration))
-
-	if autoRenew {
-		values.Set("autoRenew", "1")
-	} else {
-		values.Set("autoRenew", "0")
+// GetSmartOpenOrders gets a list of pending smart orders for an account
+func (e *Exchange) GetSmartOpenOrders(ctx context.Context, limit uint64, types []string) ([]*SmartOrder, error) {
+	params := url.Values{}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
 	}
-
-	values.Set("lendingRate", strconv.FormatFloat(rate, 'f', -1, 64))
-
-	type Response struct {
-		Success int    `json:"success"`
-		Error   string `json:"error"`
-		OrderID int64  `json:"orderID"`
+	if len(types) > 0 {
+		params.Set("types", strings.Join(types, ","))
 	}
+	var resp []*SmartOrder
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, sGetSmartOrdersEPL, http.MethodGet, "/smartorders", params, nil, &resp)
+}
 
-	result := Response{}
-
-	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, poloniexCreateLoanOffer, values, &result)
+// GetSmartOrderDetails retrieves a smart order's detail
+func (e *Exchange) GetSmartOrderDetails(ctx context.Context, orderID, clientSuppliedID string) ([]*SmartOrderDetails, error) {
+	path, err := orderPath(orderID, "/smartorders/", clientSuppliedID, "/smartorders/cid:")
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
-
-	if result.Success == 0 {
-		return 0, errors.New(result.Error)
+	var smartOrders []*SmartOrderDetails
+	resp := &V3ResponseWrapper{
+		Data: &smartOrders,
 	}
```

Comment:
🚧 Change request:
As per [previous request](https://github.com/thrasher-corp/gocryptotrader/pull/2105#discussion_r2536066425), this endpoint does not appear to be V3 wrapped.
```suggestion
```